index/generator/library/library.go

```diff
@@ -214,7 +231,11 @@ func parseExtraDevfileEntries(registryDirPath string, force bool) ([]schema.Sche
 				// Index component validation
 				err := validateIndexComponent(indexComponent, devfileType)
 				if err != nil {
-					return nil, fmt.Errorf("%s index component is not valid: %v", indexComponent.Name, err)
+					if _, ok := err.(*MissingArchError); !ok {
```

Comment:
Slight nit, take it or leave it:

This section is duplicated with the above section you added, could we abstract it out to a function?

Reply:
gonna leave it, more lines of code(for err checking) if we abstract it

Reply:
Fair enough